File: android/app/src/main/java/com/tandem/emt/features/incidentReport/models/IncidentReport.kt
package com.tandem.emt.features.incidentReport.models

data class IncidentReport(
    val id: String,
    val patientName: String,
    val patientAge: Int?,
    val chiefComplaint: String,
    val vitalSigns: VitalSigns,
    val treatments: List<String>,
    val procedures: List<String>,
    val disposition: Disposition?,
    val transportDestination: String?,
    val narrative: String,
    val timestamps: ReportTimestamps,
    val status: ReportStatus,
    val createdAt: Long,
    val updatedAt: Long
)

data class VitalSigns(
    val pulseRate: Int? = null,
    val bloodPressureSystolic: Int? = null,
    val bloodPressureDiastolic: Int? = null,
    val respirationRate: Int? = null,
    val spO2: Int? = null
)

data class ReportTimestamps(
    val arrivalTime: Long? = null,
    val patientContactTime: Long? = null,
    val transportTime: Long? = null
)

enum class Disposition(val displayName: String) {
    TRANSPORTED("Transported"),
    REFUSAL("Refusal"),
    NO_PATIENT("No Patient"),
    DEAD_ON_ARRIVAL("Dead on Arrival"),
    OTHER("Other")
}

enum class ReportStatus {
    DRAFT,
    SUBMITTED
}

File: ios/TandemEMT/Features/IncidentReport/Models/IncidentReport.swift
import Foundation

struct IncidentReport: Identifiable, Codable {
    let id: String
    var patientName: String
    var patientAge: Int?
    var chiefComplaint: String
    var vitalSigns: VitalSigns
    var treatments: [String]
    var procedures: [String]
    var disposition: Disposition?
    var transportDestination: String?
    var narrative: String
    var timestamps: ReportTimestamps
    var status: ReportStatus
    let createdAt: Date
    var updatedAt: Date
}

struct VitalSigns: Codable, Equatable {
    var pulseRate: Int?
    var bloodPressureSystolic: Int?
    var bloodPressureDiastolic: Int?
    var respirationRate: Int?
    var spO2: Int?
}

struct ReportTimestamps: Codable, Equatable {
    var arrivalTime: Date?
    var patientContactTime: Date?
    var transportTime: Date?
}

enum Disposition: String, Codable, CaseIterable {
    case transported
    case refusal
    case noPatient
    case deadOnArrival
    case other

    var displayName: String {
        switch self {
        case .transported: return "Transported"
        case .refusal: return "Refusal"
        case .noPatient: return "No Patient"
        case .deadOnArrival: return "Dead on Arrival"
        case .other: return "Other"
        }
    }
}

enum ReportStatus: String, Codable {
    case draft
    case submitted
}
